cnd/.env

```diff
@@ -0,0 +1 @@
+DATABASE_URL=cnd.sqlite3
```

Comment:
current folder?

Reply:
> > Currently we initialize the database using the `diesel` command line tool: `diesel setup`
> > This requires setting the env var DATABASE_URL, currently this is set in `cnd/.env`
> 
> @tcharding I did some quick investigation of what the `diesel` CLI acually does and there is not much magic behind it. 

Thanks man, this post is/was really helpful.

> The setup command starts here: [diesel-rs/diesel:diesel_cli/src/main.rs@`master`#L68](https://github.com/diesel-rs/diesel/blob/master/diesel_cli/src/main.rs#L68) [](/diesel-rs/diesel/blob/HEAD@{2019-08-23T00:41:50Z}/diesel_cli/src/main.rs#L68)
> 
> If you follow the code, it is mainly just creating config files and initial migrations.
> 
> I would propose the following (@comit-network/rust-devs, feel free to jump in here and disagree/argue):
> 
>     * We use the diesel CLI to manage migrations during development
> 
>     * We only support a single backend (sqlite) and hence, most of the abstractions provided by `diesel-cli` (DATABASE_URL f.e.) are not useful to us
> 
>     * `cnd` has to create the database by itself at runtime. From looking at the diesel-cli code, this can be done through `SqliteConnection::establish(path)`
> 
>     * `cnd` has to apply migrations at startup. This means we need to:
>       
>       * embed the migrations from the repository into the binary (there is a diesel macro for that)
>       * run the migrations at startup against the given database
> 
>     * We will have to create the schema at compile-time, probably through a buildscript (i.e. `build.rs`)
> 
>     * We should write a blogpost about this approach :)

Ok, will work on these points.

> 
> 
> > Binary to interact with the database is available in `cnd/src/bin/db.rs`
> 
> Since it is SQL, you can just use any SQL-viewer to view the database :)
> If it is helpful to you, feel free to have the CLI but I wouldn't merge it into master and ship it.

Cool, dropping this.